### ISSUE TITLE
BUG: StringArray is a subclass of PandasArray

### ISSUE THREAD

**Issue by ehsantn** [pandas-dev/pandas#48638]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
In [2]: A = pd.array(["a1", "a2"], dtype="string")

In [3]: A
Out[3]:
<StringArray>
['a1', 'a2']
Length: 2, dtype: string

In [4]: isinstance(A, pd.arrays.StringArray)
Out[4]: True

In [5]: isinstance(A, pd.arrays.PandasArray)
Out[5]: True
```

### Issue Description

PandasArray is described as a thin wrapper around Numpy values for internal compatibility. However, StringArray is a subclass of PandasArray for some reason (other ExtensionArrays are not). This inconsistency causes issues for unboxing the values for Bodo JIT (and I assume other systems consuming Pandas data).

https://pandas.pydata.org/docs/reference/api/pandas.Series.array.html
https://pandas.pydata.org/docs/reference/api/pandas.arrays.PandasArray.html

### Expected Behavior

Output of `isinstance(pd.array(["a1", "a2"], dtype="string"), pd.arrays.PandasArray)` should be False.

### Installed Versions

<details>
INSTALLED VERSIONS
------------------
commit           : bf856a85a3b769821809ff456e288542af16310c
python           : 3.9.6.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 21.2.0
Version          : Darwin Kernel Version 21.2.0: Sun Nov 28 20:28:54 PST 2021; root:xnu-8019.61.5~1/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.6.0.dev0+145.gbf856a85a3
numpy            : 1.21.1
pytz             : 2021.1
dateutil         : 2.8.2
setuptools       : 49.6.0.post20210108
pip              : 21.2.2
Cython           : 0.29.32
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : 7.26.0
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           :
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : None
</details>

**Comment by jbrockmendel** [comment]
> This inconsistency causes issues for unboxing the values for Bodo JIT

Can you elaborate on this?

**Comment by datapythonista** [comment]
Thanks for the report. I don't think the API for extension arrays is very clear. Did you try to simply remove the subclass [here](https://github.com/pandas-dev/pandas/blob/main/pandas/core/arrays/string_.py#L220) and see what fails? I guess there may be some generic functionality that lives in the parent classes meant for numpy backed data. It'd surely make sense to move things to the right place if that's the case.

**Comment by ehsantn** [comment]
> > This inconsistency causes issues for unboxing the values for Bodo JIT
> 
> Can you elaborate on this?

The code paths for unboxing each array type is different. For PandasArray, Bodo just needs to unbox `_ndarray` since PandasArray is just a wrapper for Numpy arrays. StringArray is a different code path with slight differences such as handling pd.NA. This led to an error in Bodo so I noticed this class hierarchy issue.

**Comment by ehsantn** [comment]
> Thanks for the report. I don't think the API for extension arrays is very clear. Did you try to simply remove the subclass [here](https://github.com/pandas-dev/pandas/blob/main/pandas/core/arrays/string_.py#L220) and see what fails? I guess there may be some generic functionality that lives in the parent classes meant for numpy backed data. It'd surely make sense to move things to the right place if that's the case.

Yes, the APIs should be clarified to enable more robust integration with Pandas in general. Removing the subclass causes an error somewhere else so it may be a bit of work. Not a big issue for us right now since the workaround is just an extra type check. Opened the issue as feedback about extension array APIs and what other users may expect.

**Comment by mroeschke** [comment]
I think this is an implementation detail that is unlikely to have guarantees so closing as no action
